jni_gen/examples/pdfbox_example/README.md

```diff
@@ -0,0 +1,34 @@
+# pdfbox_plugin_example
+
+Demonstrates how to use the PDFBox bindings generated using `jni_gen`. This is a Linux Flutter application. The sample application lists the PDF files in a directory with number of pages and title, also allowing to navigate between directories.
+
+First, it's required to have generated the dart and C bindings in the plugin directory, which also downloads the required JARs using maven. The bindings are not committed because generated code is several thousands of lines.
+
+On a Linux machine, following commands can be used to run the example application.
+
+```
+cd pdfbox_plugin ## From this folder
+dart run jni_gen --config jnigen.yaml ## Downloads PDFBox JARs and generates bindings.
+cd ..
+flutter run --release ## Opens the files list from home directory
```

Comment:
On Android emulator:

```
W/DartWorker( 3136): type=1400 audit(0.0:92): avc: denied { read } for name="/" dev="dm-4" ino=2 scontext=u:r:untrusted_app:s0:c125,c256,c512,c768 tcontext=u:object_r:rootfs:s0 tclass=dir permissive=0 app=com.example.pdfbox_plugin_example
E/flutter ( 3136): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: FileSystemException: Directory listing failed, path = './' (OS Error: Permission denied, errno = 13)
E/flutter ( 3136): 
The Flutter DevTools debugger and profiler on Android SDK built for x86 is available at:
```

On Flutter desktop:

```
$ flutter run -d linux
Launching lib/main.dart on Linux in debug mode...
Building Linux application...                                           
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: FileSystemException: Directory listing failed, path = '../mvn_jar/' (OS Error: No such file or directory, errno = 2)
#0      _Directory._fillWithDirectoryListing (dart:io-patch/directory_patch.dart:42:24)
#1      _Directory.listSync (dart:io/directory_impl.dart:243:5)
```

Should it maybe have a file dialog so you can browse to a single PDF file instead?

Reply:
It's not really supposed to run on android right now. (Have to add proguard rules to not chop off PDFBox classes but more importantly, file access on android changed a lot recently, just didn't want to deal with that.).

On linux, I forgot to test it after moving folders as Liam suggested. It should open 'pdfbox_plugin/mvn_jar' now.

Reply:
Make the app not throw an exception on Android but display a message in the UI that it's currently only supported on Flutter desktop.